src/nupic/regions/SPRegion.cpp

```diff
@@ -0,0 +1,945 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013-2018, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ *
+ * Author: David Keeney, April 2018
+ * ---------------------------------------------------------------------
+ */
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <nupic/algorithms/SpatialPooler.hpp>
+#include <nupic/engine/Input.hpp>
+#include <nupic/engine/Output.hpp>
+#include <nupic/engine/Region.hpp>
+#include <nupic/engine/Spec.hpp>
+#include <nupic/ntypes/Array.hpp>
+#include <nupic/ntypes/ArrayBase.hpp>
+#include <nupic/ntypes/BundleIO.hpp>
+#include <nupic/ntypes/Value.hpp>
+#include <nupic/regions/SPRegion.hpp>
+#include <nupic/utils/Log.hpp>
+
+#define VERSION 1 // version for streaming serialization format
+
+namespace nupic {
```

Comment:
are all of these directly in ::nupic? 
Not in this PR, but would be nice to introduce `::nupic::networkapi` namespace for these

Reply:
yes, everything is under nupic namespace.